sync/class.jetpack-sync-functions.php

```diff
@@ -142,17 +142,47 @@ public static function file_system_write_access() {
 	}
 
 	public static function home_url() {
-		return self::get_protocol_normalized_url(
-			'home_url',
-			self::normalize_www_in_url( 'home', 'home_url' )
-		);
+		if (
+			! Jetpack_Constants::is_defined( 'JETPACK_SYNC_USE_RAW_URL' ) ||
+			Jetpack_Constants::get_constant( 'JETPACK_SYNC_USE_RAW_URL' )
+		) {
+			$home_url = self::get_raw_url( 'home' );
+		} else {
+			$home_url = self::normalize_www_in_url( 'home', 'home_url' );
+		}
+
+		$home_url = self::get_protocol_normalized_url( 'home_url', $home_url );
+
+		/**
+		 * Allows overriding of the home_url value that is synced back to WordPress.com.
+		 *
+		 * @since 4.6
+		 *
+		 * @param string $home_url
+		 */
+		return esc_url_raw( apply_filters( 'jetpack_sync_home_url', $home_url ) );
 	}
 
 	public static function site_url() {
-		return self::get_protocol_normalized_url(
-			'site_url',
-			self::normalize_www_in_url( 'siteurl', 'site_url' )
-		);
+		if (
```

Comment:
as we apply the same things in both methods `home_url` and `site_url` I think we could extract this to another method